Cannot find module '/node_modules/gatsby-cli/lib/index.js'

Hi,

On a fresh install, an error is thrown when trying to serve a gatsby app, please see below.

## Steps to reproduce :
`npx create-nx-workspace gatsby` (empty)
`npm install @nrwl/gatsby --save-dev`
`nx generate @nrwl/gatsby:app myblog` (scss)
`nx serve myblog`

> nx run myblog:serve
> internal/modules/cjs/loader.js:638
>    throw err;
>    ^
>
> Error: Cannot find module '/path/to/my/nx-workspace/node_modules/gatsby-cli/lib/index.js'
>    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
>    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
>  at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
>   at startup (internal/bootstrap/node.js:283:19)
>    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
> Error during serve 


## package.json

```
{
  "name": "gatsby",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "nx": "nx",
    "start": "nx serve",
    "build": "nx build",
    "test": "nx test",
    "lint": "nx workspace-lint && nx lint",
    "e2e": "nx e2e",
    "affected:apps": "nx affected:apps",
    "affected:libs": "nx affected:libs",
    "affected:build": "nx affected:build",
    "affected:e2e": "nx affected:e2e",
    "affected:test": "nx affected:test",
    "affected:lint": "nx affected:lint",
    "affected:dep-graph": "nx affected:dep-graph",
    "affected": "nx affected",
    "format": "nx format:write",
    "format:write": "nx format:write",
    "format:check": "nx format:check",
    "update": "nx migrate latest",
    "workspace-schematic": "nx workspace-schematic",
    "dep-graph": "nx dep-graph",
    "help": "nx help"
  },
  "private": true,
  "dependencies": {
    "gatsby": "2.20.24",
    "gatsby-image": "2.2.39",
    "gatsby-plugin-svgr": "2.0.2",
    "gatsby-plugin-manifest": "2.2.39",
    "gatsby-plugin-offline": "3.0.32",
    "gatsby-plugin-react-helmet": "3.1.21",
    "gatsby-plugin-sharp": "2.4.3",
    "gatsby-source-filesystem": "2.1.46",
    "gatsby-transformer-sharp": "2.3.13",
    "prop-types": "15.7.2",
    "react": "16.12.0",
    "react-dom": "16.12.0",
    "react-helmet": "5.2.1",
    "gatsby-plugin-typescript": "2.4.0"
  },
  "devDependencies": {
    "@nrwl/gatsby": "^0.940.0-beta.0",
    "@nrwl/workspace": "9.4.5",
    "@types/node": "~8.9.4",
    "dotenv": "6.2.0",
    "eslint": "6.8.0",
    "prettier": "2.0.4",
    "ts-node": "~7.0.0",
    "tslint": "~6.0.0",
    "typescript": "~3.8.3",
    "@nrwl/jest": "9.4.5",
    "jest": "25.2.3",
    "@types/jest": "25.1.4",
    "ts-jest": "25.2.1",
    "cypress": "^4.1.0",
    "@nrwl/cypress": "9.4.5",
    "@angular-devkit/schematics": "9.1.7",
    "@nrwl/react": "9.4.0-rc.6",
    "@testing-library/react": "10.2.1",
    "babel-plugin-module-resolver": "4.0.0",
    "@nrwl/eslint-plugin-nx": "9.4.5",
    "@typescript-eslint/parser": "2.19.2",
    "@typescript-eslint/eslint-plugin": "2.19.2",
    "eslint-config-prettier": "6.0.0",
    "eslint-plugin-import": "2.20.2",
    "eslint-plugin-jsx-a11y": "6.2.3",
    "eslint-plugin-react": "7.20.0",
    "eslint-plugin-react-hooks": "4.0.2",
    "node-sass": "4.14.1",
    "gatsby-plugin-sass": "2.3.3"
  }
}
```

I tried to `npm install gatsby-cli` but it does not solve the problem.

Any idea ?

--- a/packages/gatsby-plugin/package.json
+++ b/packages/gatsby-plugin/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@nrwl/gatsby",
-  "version": "10.4.4",
+  "version": "10.4.4-beta.1",
   "description": "Gatsby Plugin for Nx",
   "repository": {
     "type": "git",
@@ -26,6 +26,7 @@
     "@nrwl/workspace": "*"
   },
   "dependencies": {
+    "gatsby-cli": "^2.14.1",
     "@nrwl/react": "^10.4.4",
     "@nrwl/cypress": "^10.4.4",
     "@nrwl/jest": "^10.4.4"

--- a/packages/gatsby-plugin/src/builders/build/builder.ts
+++ b/packages/gatsby-plugin/src/builders/build/builder.ts
@@ -37,7 +37,7 @@ export function runGatsbyBuild(
 ) {
   return new Promise((resolve, reject) => {
     const cp = fork(
-      join(workspaceRoot, './node_modules/gatsby-cli/lib/index.js'),
+      require.resolve('gatsby-cli'),
       ['build', ...createGatsbyBuildOptions(options)],
       {
         cwd: join(workspaceRoot, `apps/${project}`),

--- a/packages/gatsby-plugin/src/builders/server/builder.ts
+++ b/packages/gatsby-plugin/src/builders/server/builder.ts
@@ -79,17 +79,13 @@ function createGatsbyOptions(options) {
 
 async function runGatsbyDevelop(workspaceRoot, project, options) {
   return new Promise<boolean>((resolve, reject) => {
-    const cp = fork(
-      join(workspaceRoot, './node_modules/gatsby-cli/lib/index.js'),
-      ['develop', ...options],
-      {
-        cwd: join(workspaceRoot, `apps/${project}`),
-        env: {
-          ...process.env,
-        },
-        stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
-      }
-    );
+    const cp = fork(require.resolve('gatsby-cli'), ['develop', ...options], {
+      cwd: join(workspaceRoot, `apps/${project}`),
+      env: {
+        ...process.env,
+      },
+      stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
+    });
 
     // Ensure the child process is killed when the parent exits
     process.on('exit', () => cp.kill());
@@ -125,7 +121,7 @@ function runGatsbyServe(
     const cwd = join(workspaceRoot, `apps/${project}`);
 
     const cp = fork(
-      join(workspaceRoot, './node_modules/gatsby-cli/lib/index.js'),
+      require.resolve('gatsby-cli'),
       ['serve', ...createGatsbyServeOptions(options)],
       { cwd }
     );

